CONTRIBUTING.md

```diff
@@ -29,21 +29,47 @@ Qt:
   will (very likely) be **ignored**. :fearful:
 
 
-# Opening a pull request
+# How to start a pull request
 
-### Must read
+## Must read
+* Before you start contributing, first decide for a specific topic. Pull requests, which are spanning multiple topics (e.g. "general qTox code cleanup") or introduce fundamental architectural changes are rare and require additional attention and maintenance.
 * Use [**commit message format**](#commit-message-format).
 * Read our [**coding guidelines**](#coding-guidelines).
 * Keep the title **short** and provide a **clear** description about what your
   pull request does.
 * Provide **screenshots** for UI related changes.
-* Keep your git commit history **clean** and **precise**. Commits like
-  `xxx fixup` should not appear.
+* Keep your git commit history **clean** and **precise** by continuously rebasing/amending your PR. Commits like `xxx fixup` are not needed and rejected during review.
 * If your commit fixes a reported issue (for example #4134), add the following
   message to the commit `Fixes #4134.`.
   [Here is an example](https://github.com/tux3/qTox/commit/87160526d5bafcee7869d6741a06045e13d731d5).
 
-### Good to know
+## How to avoid merge conflicts
+If you experience big amounts of conflicts, they are indicating a problem in your communication and appear e.g. on PR's, that try to "fix everything". So the best way is to completely avoid "hard-to-solve" merge conflicts by concentrating on smaller related chunks of code. Anyway, if you run into such situation, here's a "Rule of Thumb" list to get out of it:
+
+1. Split your commit history into topic related chunks (by rebasing them).
+2. Split "API" and "UI" (widget related) changes into separate commits.
+3. Probably split PR into several smaller ones.
+
+In addition it helps to regularly keep rebasing on the main repository's recent master. You can do this by setting up a second Git remote as follows:
+
+~~~bash
+# Add a Git remote for the qTox main repository:
+git remote add main https://github.com/tux3/qTox.git
+
+# If not on PR branch, check it out:
+git checkout my/pr-branch
+
+# Now fetch master ALWAYS from main repo
+git fetch main master:master
+
+# Last, rebase PR branch onto master…
+git rebase -i master
+
+# …and, if everything's clear, force push to YOUR repo (your "origin" Git remote)
+git push -f
```

Comment:
Git claims otherwise:
```
$ git push -f
warning: push.default is unset; its implicit value has changed in
Git 2.0 from 'matching' to 'simple'. To squelch this message
and maintain the traditional behavior, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

When push.default is set to 'matching', git will push local branches
to the remote branches that already exist with the same name.

Since Git 2.0, Git defaults to the more conservative 'simple'
behavior, which only pushes the current branch to the corresponding
remote branch that 'git pull' uses to update the current branch.

See 'git help config' and search for 'push.default' for further information.
(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode
'current' instead of 'simple' if you sometimes use older versions of Git)

fatal: The current branch test has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin test
```